File: ClientAPI/src/main/java/pt/upskill/clientapi/DTOs/GreenitorDTO.java
package pt.upskill.clientapi.DTOs;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

/**
 * Data Transfer Object (DTO) for the Greenitor entity.
 * This class represents the data associated with a Greenitor, including their username, email, role, interactions, image, and badges.
 */
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class GreenitorDTO {

    /**
     * Username of the Greenitor.
     * This field stores the unique identifier for the Greenitor.
     */
    private String username;

    /**
     * Email of the Greenitor.
     * This field contains the email address associated with the Greenitor account.
     */
    private String email;

    /**
     * Role of the Greenitor.
     * This field defines the role (e.g., "Admin", "User") of the Greenitor within the application.
     */
    private String role;

    /**
     * Interactions of the Greenitor.
     * This field holds the number of interactions the Greenitor has made within the system.
     */
    private int interactions;

    /**
     * Image of the Greenitor.
     * This field contains the URL or path to the Greenitor's profile image.
     */
    private String image;

    /**
     * Badges achieved by the Greenitor.
     * This field holds a list of badges the Greenitor has earned, represented as BadgeDescriptionDTO objects.
     */
    private List<BadgeDescriptionDTO> badges;
}

File: ForumAPI/ForumAPI/DTOs/GreenitorDTOs/GreenitorDTO.cs
using ForumAPI.DTOs.BadgeDTOs;

namespace ForumAPI.DTOs.GreenitorDTOs
{
    /// <summary>
    /// Data Transfer Object (DTO) for a greenitor, which includes the greenitor's username, email, role, interactions, image, and associated badges.
    /// </summary>
    public class GreenitorDTO
    {
        /// <summary>
        /// Gets or sets the username of the greenitor.
        /// </summary>
        public string Username { get; set; }

        /// <summary>
        /// Gets or sets the email address of the greenitor.
        /// </summary>
        public string Email { get; set; }

        /// <summary>
        /// Gets or sets the role of the greenitor, such as 'Admin', 'Moderator', or 'User'.
        /// </summary>
        public string Role { get; set; }

        /// <summary>
        /// Gets or sets the number of interactions the greenitor has made (e.g., posts, comments, likes).
        /// </summary>
        public int Interactions { get; set; }

        /// <summary>
        /// Gets or sets the URL or file path to the greenitor's profile image.
        /// </summary>
        public string Image { get; set; }

        /// <summary>
        /// Gets or sets the list of badges associated with the greenitor.
        /// Each badge is represented by a BadgeDescriptionDTO.
        /// </summary>
        public List<BadgeDescriptionDTO> Badges { get; set; }

        /// <summary>
        /// Initializes a new instance of the <see cref="GreenitorDTO"/> class.
        /// </summary>
        /// <param name="username">The username of the greenitor.</param>
        /// <param name="email">The email address of the greenitor.</param>
        /// <param name="interactions">The number of interactions the greenitor has made.</param>
        public GreenitorDTO(string username, string email, int interactions)
        {
            Username = username;
            Email = email;
            Interactions = interactions;
        }

        public GreenitorDTO()
        {
        }
    }
}
